R/utils-read.R

```diff
@@ -308,27 +308,34 @@ readApprovalPoints <- function(approvals, points, timezone, legend_nm, appr_var_
   
   date_index_list <- list(list(type="Approved",approved_index), list(type="In Review",review_index), list(type="Working",working_index))
   
-  for(sub_list in date_index_list){
-    approval_info <- list()
-    for(list in sub_list){
-      appr_var <- appr_var_all[which(appr_type == sub_list["type"])]
-      for(i in seq_along(list)){
-        d <- list[[i]]
-        
-        applicable_dates <- points[['time']][d]
-        applicable_values <- points[['value']][d]
-        
-        approval_info[[i]] <- list(time = applicable_dates,
-            value = applicable_values,
-            legend.name = paste(sub_list["type"], legend_nm),
-            point_type = point_type)
-      }
+  for(level in date_index_list){
+    
+    appr_var <- appr_var_all[which(appr_type == level["type"])]
+    list <- level[[2]]
+    
+    approval_info <- data.frame(time=.POSIXct(character()),
```

Comment:
interesting...because I was getting an error in some scenarios without the `as.`